aiohttp/web_request.py

```diff
@@ -444,8 +448,10 @@ def remote(self) -> Optional[str]:
 
     @reify
     def url(self) -> URL:
-        url = URL.build(scheme=self.scheme, host=self.host)
-        return url.join(self._rel_url)
+        """The full URL of the request."""
+        # authority is used here because it may include the port number
+        # and we want yarl to parse it correctly
+        return URL.build(scheme=self.scheme, authority=self.host).join(self._rel_url)
```

Comment:
Sadly this is ~63x slower when passing `authority`

Reply:
We already have a fast helper in yarl for that so we can get it to be much faster https://github.com/aio-libs/yarl/pull/1163